pages/api/pearl-daa.js

```diff
@@ -0,0 +1,214 @@
+      {
+        agentId_in: traderAgentIds,
+        dayTimestamp_gte: timestamp_gte,
+        dayTimestamp_lte: timestamp_lte,
+      },
+    );
+
+    return performance?.dailyAgentPerformances ?? [];
+  } catch (error) {
+    console.error('GraphQL query error:', error);
+    return [];
+  }
+};
+
+const groupServicesByDays = (activeServices, predictServiceIds) => {
+  const servicesByDays = {};
+
+  for (const service of activeServices) {
+    const timestamp = service.dayTimestamp;
+
+    if (!servicesByDays[timestamp]) {
+      servicesByDays[timestamp] = new Set();
+    }
+
+    for (const multisig of service.multisigs) {
+      const serviceId = Number(multisig.multisig.serviceId);
+      if (predictServiceIds.has(serviceId)) {
+        servicesByDays[timestamp].add(multisig.multisig.id);
+      }
+    }
+  }
+
+  return servicesByDays;
+};
+
+const getBabydegenOptimusDailyCounts = async () => {
+  const timestamp_lt = getMidnightUtcTimestampDaysAgo(0);
+  const timestamp_gt = getMidnightUtcTimestampDaysAgo(7);
+
+  try {
+    const result = await REGISTRY_GRAPH_CLIENTS.optimism.request(
+      dailyBabydegenPerformancesQuery,
+      {
+        timestamp_gt,
+        timestamp_lt,
+      },
+    );
+
+    const performances = result.dailyAgentPerformances ?? [];
+    const dailyCounts = {};
+
+    performances.forEach((perf) => {
+      const timestamp = perf.dayTimestamp;
+      dailyCounts[timestamp] = perf.activeMultisigCount;
+    });
+
+    return dailyCounts;
+  } catch (error) {
+    console.error('Error fetching Optimism DAAs:', error);
+    return {};
+  }
+};
+
+const getCombinedPearlDAAs = async () => {
+  const [predictResult, babydegenResult] = await Promise.allSettled([
+    getPredictDailyCounts(),
+    getBabydegenOptimusDailyCounts(),
+  ]);
+
+  const predictDailyCounts =
+    predictResult.status === 'fulfilled' ? predictResult.value : {};
+  const babydegenDailyCounts =
+    babydegenResult.status === 'fulfilled' ? babydegenResult.value : {};
+
+  const allTimestamps = new Set([
+    ...Object.keys(predictDailyCounts),
+    ...Object.keys(babydegenDailyCounts),
+  ]);
+
+  const sortedTimestamps = Array.from(allTimestamps).sort(
+    (a, b) => Number(a) - Number(b),
+  );
+
+  const combinedCounts = {};
+  const dailyCountsArray = [];
+
+  sortedTimestamps.forEach((timestamp) => {
+    const predictCount = predictDailyCounts[timestamp] || 0;
+    const babydegenCount = babydegenDailyCounts[timestamp] || 0;
+    const combined = predictCount + babydegenCount;
+
+    // Convert to date string for output readability
+    const dateString = new Date(Number(timestamp) * 1000)
+      .toISOString()
+      .slice(0, 10);
+
+    combinedCounts[dateString] = {
+      predict: predictCount,
+      babydegen: babydegenCount,
+      total: combined,
+    };
+    dailyCountsArray.push({ count: combined });
+  });
+
+  const averageDAAs = calculate7DayAverage(dailyCountsArray, 'count');
```

Comment:
I think we're doing wrong math here.
we have to get 7 days average DAA for predict and separately for babydegen. then sum up the averages.
we also shouldn't sort timestamps and make some new dateString etc., this is confusing and not needed.
as well as we shouldn't do `groupServicesByDays`, we already have them by days at least for babydegen count. for predict we only need to update total counter and keep there only pearl contracts from checkpoints.

please let's make it simple
```suggestion
  const babydegenAverage = calculate7DayAverage(dailyAgentPerformances, 'activeMultisigCount'); // dailyAgentPerformances are just performances from getBabydegenOptimusDailyCounts
  const predictAverage = calculate7DayAverage(dailyAgentPerformances, 'activeMultisigCount'); // dailyAgentPerformances is updated based on the logic where we should exclude non-pearl
  
  const averageDAAs = babydegenAverage + predictAverage;
```

you can see an example here as well: https://github.com/valory-xyz/olas-website/blob/main/pages/api/babydegen-metrics.js#L335-L344